torch/lib/THD/benchmark/run_benchmark

```diff
@@ -0,0 +1,163 @@
+#! /bin/sh
+
+set -eu
+
+BACKEND=mpi
+engine="$PWD/benchmark.py"
+environment=/dev/null
+master_hostname=localhost
+output_file=/dev/stdout
+hosts=localhost
+MASTER_PORT=29500
+MASTER_ADDR="$master_hostname:$MASTER_PORT"
+WORLD_SIZE=2
+
+errxit() {
+    printf "%s\n" "$*" 1>&2
+    exit 1
+}
+
+usage() {
+    cat <<-EOF
+Usage: ./run_benchmark [ OPTIONS ]
+
+Optional arguments:
+    -., --env FILE
+        Set the path to a file to source. When using the MPI backend, the file
+        will be sourced before running 'mpirun'. In case of the TCP backend
+        the file will be sourced on every host after establishing a successful
+        SSH connection. Default: '/dev/null'.
+
+    -b, --backend BACKEND
+        Set the backend to benchmark. Default: 'mpi'.
+
+    -e, --engine ENGINE
+        Set the path to the benchmarking script to run. Use absolute paths if
+        you'll be using the tcp backend. Default: '\$PWD/benchmark.py'.
+
+    --help
+        Show this help and exit successfully.
+
+    -h, --hosts HOSTS
+        Set the list of hosts to run the benchmark on. Format: 'host1,host2'.
+        Default: 'localhost'.
+
+    --max-bytes MAX_BYTES
+        Set the inclusive upper limit for tensor size.
+        Default: 22 (2**22 = 4 MB).
+
+    --max-num-tensors MAX_NUM_TENSORS
+        Set the inclusive upper limit for the number of tensors to be sent
+        during one test run. Default: 3 (10**3 = 1000).
+
+    --min-bytes MIN_BYTES
+        Set the inclusive lower limit for tensor bytes.
+        Default: 19 (2**19 = 512 KB).
+
+    --min-num-tensors MIN_NUM_TENSORS
+        Set the inclusive upper limit for the number of tensors to be sent
+        during one test run. Default: 2 (10**2 = 100).
+
+    -n, --name NAME
+        Set the ip address/host name of the master node. Default: 'localhost'.
+
+    -o, --output FILE
+        Set the path to the output file where the master host will append
+        benchmark results. Default: '/dev/stdout'.
+
+    -p, --port PORT
+        Set the port number master is listening on. Default: '29500'.
+
+    -s, --world-size WORLD_SIZE
+        Set the number of processes to be spawned. Default: '2'.
+EOF
+}
+
+while [ $# -gt 0 ]; do
```

Comment:
`set -u` is on during the whole script so I guess I am not going to add any additional check. Good point though!